core/shared/src/test/scala/zio/prelude/TraversableSpec.scala

```diff
@@ -115,6 +124,49 @@ object TraversableSpec extends DefaultRunnableSpec {
             assert(actual)(equalTo(expected))
           }
         },
+        testM("forany") {
+          check(genList, genOptIntFunction) { case (as, f) =>
+            val actual   = Traversable[List].forany(as)(f).flatten
+            val expected = as.map(f).flatten.headOption
+            assert(actual)(equalTo(expected))
+          }
+        },
+        testM("forany IO") {
+          check(genList, genOptIntFunction) { case (as, f) =>
+            val actual   = zio.Runtime.default.unsafeRun(Traversable[List].forany(as)(f.map {
+              case Some(x) => IO.succeedNow(x)
+              case None => IO.fail(())
+            }(Invariant.Function1Covariant)).option)
+            val expected : Option[Option[Int]] = {
+              val allResults = as.map(f)
+              if(allResults.nonEmpty && allResults.forall(_.isEmpty))
+                None
+              else
+                Some(allResults.flatten.headOption)
+            }
+            assert(actual)(equalTo(expected))
+          }
+        },
+        testM("foranyPar IO") {
+          check(genList, genOptIntFunction) { case (as, f) =>
+            val actual = zio.Runtime.default.unsafeRun(Traversable[List].foranyPar(as)(f.map {
+              case Some(x) => IO.succeedNow(x)
+              case None => IO.fail(())
+            }(Invariant.Function1Covariant)).option)
+            val expected : Option[Set[Int]] = {
+              val allResults = as.map(f)
+              if(allResults.nonEmpty && allResults.forall(_.isEmpty))
+                None
+              else
+                Some(allResults.flatten.toSet)
+            }
+            (actual, expected) match {
+              case (None , None)                            => assertCompletes
+              case (Some(Some(result)), Some(allPossible))  => assert(List(result))(hasOneOf(allPossible))
+              case (Some(None),         Some(allPossible))  => assert(allPossible)(isEmpty)
+            }
```

Comment:
don't know how to make more informative failed assertion for exhaustive case here